Paladin Protection increased vanq hammer prio and HOW prio in ashen

xizbow1 opened the pull request:
I increased the prio of vanq hammer, and added a conditional to only cast it with seraphim if its talented and wearing DBG. I added a HOW action when in ashen with MP/VM

Commit:
better on 1, 3, 5, worse everywhere else

Commit:
reordered actions, increasing the prio of vanq hammer and only casting AH with seraphim. WIP

xizbow1 committed:
reordering actions

increased vanq hammer prio, only cast ashen with seraphim. WIP

xizbow1 committed:
increased vanq hammer prio and increased HOW prio with ashen

moved vanq hammer up the list and only casts with seraphim if using DBG, casts on cd otherwise. added a HOW action with ashen and MP/VM active

xizbow1 committed:
Update T27_Paladin_Protection.simc

xizbow1 committed:
Merge branch 'simulationcraft:shadowlands' into shadowlands

Commit:
increased vanq hammer prio and prio how in ashen

Commit:
Merge branch 'shadowlands' of https://github.com/blculbertson/simc into shadowlands

scamille commented:
The only thing problematic here is that the profile files themselves are automatically generated. The actual action priority list lives at https://github.com/simulationcraft/simc/blob/faceea27325bf85109134870961ebecd5e5031bd/engine/class_modules/paladin/sc_paladin_protection.cpp#L914 , see https://github.com/simulationcraft/simc/blob/shadowlands/CONTRIBUTING.md#action-priority-list-apl

Either you can include your changes there, or one of the paladin maintainers can pick it up. @OrcinusDrake @mserrano @Woliance or @Melekus

xizbow1 committed:
Merge branch 'simulationcraft:shadowlands' into shadowlands

xizbow1 committed:
Update T27_Paladin_Holy.simc

xizbow1 committed:
added changes to class module

as previously stated, increased vanq hammer prio and HOW prio in ashen. removed ashen with seraphim change, as it was pointed out that it is a slight loss

xizbow1 commented:
I added the changes to the class module, with some (quite a bit of) help from Woliance.

scamille commented on profiles/Tier27/T27_Paladin_Protection.simc:
```diff
@@ -50,16 +50,17 @@ actions.cooldowns+=/moment_of_glory,if=prev_gcd.1.avengers_shield&cooldown.aveng
 actions.standard+=/divine_toll
 actions.standard+=/blessed_hammer,strikes=2.4,if=charges=3
-actions.standard+=/ashen_hallow
+actions.standard+=/ashen_hallow,if=buff.seraphim.up|!talent.seraphim.enabled
```
This change does not seem to be in the cpp file

xizbow1 replied:
Yes, I reverted that change in the profile due to it being slightly incorrect play, and forgot to commit it.

scamille commented on engine/class_modules/paladin/sc_paladin_protection.cpp:
```diff
@@ -956,12 +956,13 @@ void paladin_t::generate_action_prio_list_prot()
   std -> add_action( this, "Judgment", "target_if=min:debuff.judgment.remains,if=charges=2|!talent.crusaders_judgment.enabled" );
   std -> add_action( this, "Hammer of Wrath" );
   std -> add_action( "blessing_of_the_seasons" );
+  std->add_action( "vanquishers_hammer", "if = buff.seraphim.up & runecarving.duty_bound_gavel | !runecarving.duty_bound_gavel | !talent.seraphim.enabled " );
```
```suggestion
  std -> add_action( "vanquishers_hammer", "if = buff.seraphim.up & runecarving.duty_bound_gavel | !runecarving.duty_bound_gavel | !talent.seraphim.enabled" );
```

xizbow1 committed:
respaced to look better

Co-authored-by: Serge Camille <sergecamille@gmail.com>

xizbow1 committed:
Update T27_Paladin_Protection.simc

xizbow1 committed:
Merge branch 'simulationcraft:shadowlands' into shadowlands

scamille commented on engine/class_modules/paladin/sc_paladin_protection.cpp:
```diff
@@ -956,12 +956,13 @@ void paladin_t::generate_action_prio_list_prot()
   std -> add_action( this, "Judgment", "target_if=min:debuff.judgment.remains,if=charges=2|!talent.crusaders_judgment.enabled" );
   std -> add_action( this, "Hammer of Wrath" );
   std -> add_action( "blessing_of_the_seasons" );
+  std -> add_action( "vanquishers_hammer", "if = buff.seraphim.up & runecarving.duty_bound_gavel | !runecarving.duty_bound_gavel | !talent.seraphim.enabled" );
```
```suggestion
  std -> add_action( "vanquishers_hammer", "if=buff.seraphim.up&runecarving.duty_bound_gavel|!runecarving.duty_bound_gavel|!talent.seraphim.enabled" );
```
Not sure why I didn't notice the whitespace before.

scamille replied:
I actually don't know what would happen on profile regeneration if you left the whitespace in there. But I fear things would break, so better to remove them :-)

xizbow1 committed:
more whitespace fixes at scamille's suggestion

better safe than sorry

Co-authored-by: Serge Camille <sergecamille@gmail.com>

xizbow1 committed:
Merge branch 'simulationcraft:shadowlands' into shadowlands

xizbow1 committed:
Update sc_paladin_holy.cpp

xizbow1 committed:
Merge pull request #2 from xizbow1/xizbow1-necro-holy

Paladin holy various apl updates relating to vanq hammer

xizbow1 commented:
Closing this pull request as my inexperience with github has led to it being a mess. Will open a new one without a billion separate commits.
